sdk/src/main/java/com/deploygate/sdk/DeployGate.java

```diff
@@ -127,6 +138,48 @@ public void onEvent(
                 } else {
                     Logger.w("streamed logcat is not supported");
                 }
+            } else if (DeployGateEvent.ACTION_COLLECT_DEVICE_STATES.equals(action)) {
+                Uri targetUri = Uri.parse(extras.getString(DeployGateEvent.EXTRA_TARGET_URI_FOR_REPORT_DEVICE_STATES));
+                Logger.d("collect-device-status event received: %s", targetUri);
+
+                ContentValues cv = new ContentValues();
+                synchronized (getBuildEnvironment().attributes) {
+                    if (!getBuildEnvironment().attributes.isEmpty()) {
+                        String buildEnvironmentJson = new JSONObject(getBuildEnvironment().attributes).toString();
+                        cv.put(DeployGateEvent.ATTRIBUTE_KEY_BUILD_ENVIRONMENT, buildEnvironmentJson);
+                    }
```

Comment:
What makes you check if it's empty? Can you explain what will happen in this synchronized block?

It looks a meaningless synchronized block because an empty CustomAttributes always produces the same result, i.e. an empty json object, with the optimized synchronization via *ConcurrentHashMap*. So, you should just check if the json string is `{}` or not.